### PR TITLE
Task/insert upsert transaction

### PR DESCRIPTION
**Pull request by IvanHdzC**
This PR solves issues #1806 and  #1936 for MySQL sink.

### CODE REVIEW

File: CHANGES_NEXT_RELEASE
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:2]: Maybe both lines could be joined:

```suggestion
[cygnus-ngsi, cygnus-common][PosgtgreSQLSink, PostgisSink, MySQLSQLSink] Create upsert transaction (#1806, #1936)
```

Comment (IvanHdzC) [reply CHANGES_NEXT_RELEASE]: I agreee.

Fixed in [803356b](https://github.com/telefonicaid/fiware-cygnus/pull/1944/commits/803356bc6288c2273ffb39529151d556fb54ba58)

File: doc/cygnus-ngsi/flume_extensions_catalogue/last_data_function.md
Comment (fgalan) [inline doc/cygnus-ngsi/flume_extensions_catalogue/last_data_function.md:97]: Should last_data_function.md linked from mysql sink documentation (as Postgresql/PostGIS sink doc I think do)?

Comment (IvanHdzC) [reply doc/cygnus-ngsi/flume_extensions_catalogue/last_data_function.md:97]: Fixed in [3a6ef257](https://github.com/telefonicaid/fiware-cygnus/pull/1944/commits/3a6ef257335bcb2c419b38e841d7a683dfeabe76)

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/SQLQueryUtils.java
Comment (AlvaroVega) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/SQLQueryUtils.java:58]: not sure if indentation is fine.

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/SQLQueryUtils.java
Comment (AlvaroVega) [inline cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/SQLQueryUtils.java:223]: Not sure if indentation is right

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/sql/SQLQueryUtilsTest.java
Comment (AlvaroVega) [inline cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/sql/SQLQueryUtilsTest.java:21]: Should this new dependence be added to pom.xml ?

File: doc/cygnus-ngsi/flume_extensions_catalogue/last_data_function.md
Comment (fgalan) [reply doc/cygnus-ngsi/flume_extensions_catalogue/last_data_function.md:97]: What about a link also in doc/cygnus-ngsi/flume_extensions_catalogue/README.md ?

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/SQLQueryUtils.java
Comment (IvanHdzC) [reply cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/SQLQueryUtils.java]: Fixed in [e0aa884](https://github.com/telefonicaid/fiware-cygnus/pull/1944/commits/e0aa8847b5596ace1a4bf45340f279f7aa8e90e0)

File: cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/SQLQueryUtils.java
Comment (IvanHdzC) [reply cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/sql/SQLQueryUtils.java]: Fixed in [e0aa884](https://github.com/telefonicaid/fiware-cygnus/pull/1944/commits/e0aa8847b5596ace1a4bf45340f279f7aa8e90e0)

File: cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/sql/SQLQueryUtilsTest.java
Comment (IvanHdzC) [reply cygnus-common/src/test/java/com/telefonica/iot/cygnus/backends/sql/SQLQueryUtilsTest.java]: Not really. Not sure why I added it, may be an auto suggestion from IntelliJ. Anyway, I removed it.

Fixed in [e0aa884](https://github.com/telefonicaid/fiware-cygnus/pull/1944/commits/e0aa8847b5596ace1a4bf45340f279f7aa8e90e0)

**Review by AlvaroVega — APPROVED** [review]
LGTM

File: doc/cygnus-ngsi/flume_extensions_catalogue/last_data_function.md
Comment (IvanHdzC) [reply doc/cygnus-ngsi/flume_extensions_catalogue/last_data_function.md:97]: My mistake.

Fixed in [e1fc9bf](https://github.com/telefonicaid/fiware-cygnus/pull/1944/commits/e1fc9bf8fa00baec03b7aaba267b440468a6ce71).

**Review by fgalan — APPROVED** [review]
LGTM

Should we merge or wait for additional review from @manucarrace ?

**Pull request merged**
